account_invoice_tax_required/models/account_invoice.py

```diff
@@ -28,14 +14,15 @@ def test_invoice_line_tax(self):
         errors = []
```

Comment:
We should ignore this test on 2 cases:

* With demo data. You can see the problem on current runbot build:
<details>
No Taxes Defined!
 Invoice has a line with product Dell Inspiron Laptop without OS with no taxes,Invoice has a line with product Linutop with no taxes
None
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/tools/yaml_import.py", line 884, in process
    self._process_node(node)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/tools/yaml_import.py", line 916, in _process_node
    self.process_workflow({node: []})
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/tools/yaml_import.py", line 655, in process_workflow
    openerp.workflow.trg_validate(uid, workflow.model, id, workflow.action, self.cr)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/__init__.py", line 67, in trg_validate
    return WorkflowService.new(cr, uid, res_type, res_id).validate(signal)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/service.py", line 73, in validate
    res2 = wi.validate(signal)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/instance.py", line 62, in validate
    wi.process(signal=signal, force_running=force_running, stack=stack)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 102, in process
    ok = self._split_test(activity['split_mode'], signal, stack)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 230, in _split_test
    self._join_test(t[0], t[1], stack)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 239, in _join_test
    WorkflowItem.create(self.session, self.record, activity, inst_id, stack=stack)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 77, in create
    workflow_item.process(stack=stack)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 98, in process
    if not self._execute(activity, stack):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 144, in _execute
    returned_action = self.wkf_expr_execute(activity)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 295, in wkf_expr_execute
    return self.wkf_expr_eval_expr(activity['action'])
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/workflow/workitem.py", line 273, in wkf_expr_eval_expr
    result = eval(line, env, nocopy=True)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/tools/safe_eval.py", line 298, in safe_eval
    return eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/addons/account_invoice_tax_required/models/account_invoice.py", line 31, in invoice_validate
    self.test_invoice_line_tax()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3155336-404-bea4e8/openerp/addons/account_invoice_tax_required/models/account_invoice.py", line 24, in test_invoice_line_tax
    _('No Taxes Defined!\n %s') % (errors_full_string)
UserError: (u'No Taxes Defined!\n Invoice has a line with product Dell Inspiron Laptop without OS with no taxes,Invoice has a line with product Linutop with no taxes', None)

</details>
  I'm not sure right now how can we make this...
* When making tests. We should mask the constraint on test mode for the rest of the modules. You can see an example of the inhibit here: https://github.com/OCA/bank-payment/blob/8.0/account_payment_transfer_reconcile_batch/models/payment_order.py#L28 (it contains another condition more in this case as the same method can be called from another path, but essentially it's the same), and here the way to make correctly the tests: https://github.com/OCA/bank-payment/blob/8.0/account_payment_transfer_reconcile_batch/tests/test_account_payment_transfer_reconcile_batch.py#L63